src/diffusers/models/resnet.py

```diff
@@ -373,59 +409,91 @@ def __init__(
         conv_shortcut=False,
         dropout=0.0,
         temb_channels=512,
-        groups=32,
+        groups: Optional[int] = 32,
         groups_out=None,
         pre_norm=True,
         eps=1e-6,
         non_linearity="swish",
-        time_embedding_norm="default",
+        norm1_scale_shift: bool = False,  # if true, will use ada groupnorm (i.e. time embedding scale shift) instead of groupnorm
+        time_embedding_norm="default",  # default, scale_shift
         kernel=None,
         output_scale_factor=1.0,
         use_in_shortcut=None,
         up=False,
         down=False,
+        mid_channels=None,
+        dropout_after_conv: bool = False,  # if true, will place a dropout after each conv2d
+        group_size: Optional[int] = None,  # only effective when groups is None
```

Comment:
Can we try to not add this parameter - I think we should be able to handle everything via the `groups` arg no?

Reply:
so `groups = num_channels/group_size`, if `num_channels` is known, we can calculate `groups` based on desired `group_size` - but in latent upscaler's resnet block it use the ada group norm layer twice with different number of channels, so we probably can't get the same result with just one `groups` argument

but I think we can run tests to see this argument make a difference in the upscaled image (I will do that last though)